tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/peripheral.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/bluetooth.h"
+#include "zephyr/bluetooth/conn.h"
+#include "zephyr/toolchain/gcc.h"
+
+#include <stdint.h>
+#include <string.h>
+
+void peripheral_procedure(void)
+{
+	bs_bt_utils_setup();
+
+	printk("DUT start\n");
+	k_msleep(1000);
+
+	/* start scanning (using NRPA) */
+	scan_connect_to_first_result();
+	k_msleep(100);
+
+	advertise_connectable(0, 0);
+	wait_connected();
+	printk("DUT is peripheral\n");
+
+	/* tester advertises using a new identity
+	 * -> will get detected and connected to by DUT
+	 */
+	wait_connected();
+	printk("DUT is central & peripheral\n");
+
+	/* restart advertiser: it will fail because we have run out of contexts.
+	 * But since we pass the `persist` flag, it will start up as soon as a
+	 * peripheral role is disconnected.
+	 *
+	 * We can't start it with the `persist` flag the first time, because adv
+	 * will resume straight after the peripheral's connection completes,
+	 * 'stealing' the last conn context and preventing the scanner from
+	 * establishing a connection.
+	 */
+	advertise_connectable(0, 1);
+
+	wait_disconnected();
+	printk("DUT is central\n");
+	scan_connect_to_first_result();
+
+	wait_disconnected();
+	printk("DUT has no connections\n");
+
+	k_sleep(K_SECONDS(1));
```

Comment:
Why is this sleep needed?

Reply:
so the advertiser has the time to re-start and send out a few packets after the disconnect has been fully handled. See https://github.com/zephyrproject-rtos/zephyr/pull/52940#discussion_r1049302068 .

Reply:
I see. For advertising it is a bit more tricky to handle, as we don't get events in the host when advertising packets are sent. 

Two non-timing solutions:
1) Use the babblesim backchannel to sync between the central and peripheral, so that we instead of waiting for a timer, we wait for the central to have seen our advertisements
2) Simply remove; since we are not disabling Bluetooth after `PASS`ing, the peripheral should still continue advertising after ending this function. The `k_sleep` here is basically providing nothing IMO.